### PR TITLE
chore: Remove unneeded cross feature dependency

### PR DESCRIPTION
**Pull request by ktff**
Ref. #1940 

Some feature dependencies were a leftover, and some were gating shared code so those were removed.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -261,13 +261,13 @@ sources-journald = []
...
-sources-socket = ["bytesize", "listenfd", "tokio/uds", "tokio-util/udp", "sources-tls"]
+sources-socket = ["bytesize", "listenfd", "tokio-util/udp", "sources-tls"]
 sources-splunk_hec = ["bytesize", "warp", "sources-tls"]
 sources-statsd = ["tokio-util/udp"]
 sources-stdin = ["bytesize"]
-sources-syslog = ["sources-socket", "syslog_loose"]
-sources-tls = ["sources-http", "sources-logplex", "sources-socket", "sources-splunk_hec"]
-sources-vector = ["sources-socket"]
+sources-syslog = ["bytesize", "listenfd", "tokio-util/udp", "sources-tls", "syslog_loose"]
+sources-tls = []
```
Comment (fanatid) [inline Cargo.toml:269]: Looks like this solve my question https://github.com/timberio/vector/pull/3188#issuecomment-663712811

File: Cargo.toml [hunk 2]
```diff
@@ -43,7 +43,7 @@ tracing-limit = { path = "lib/tracing-limit" }
 futures01 = { package = "futures", version = "0.1.25" }
 futures = { version = "0.3", default-features = false, features = ["compat", "io-compat"] }
 tokio01 = { package = "tokio", version = "0.1.22", features = ["io", "tcp", "rt-full", "experimental-tracing", "sync"], default-features = false }
-tokio = { version = "0.2.13", features = ["blocking", "fs", "sync", "macros", "test-util", "rt-core", "io-std"] }
+tokio = { version = "0.2.13", features = ["blocking", "fs", "sync", "macros", "test-util", "rt-core", "io-std", "uds"] }
```
Comment (fanatid) [inline Cargo.toml:46]: Why move `uds` feature from components to dependency?

File: Cargo.toml (hunk 1 above)
Comment (ktff) [reply Cargo.toml:269]: Yep.

File: Cargo.toml (hunk 2 above)
Comment (ktff) [reply Cargo.toml:46]: Our `unix` util requires this feature, and without moving it here the `unix` util  would need to be feature guarded by  sources that use it, and in turn they would need to depend on features that `unix` util needs which kinda backwards, so this is a compromise. 

An alternative would be to have separate feature for `unix` util.

**Pull request merged**